napari/components/layerlist.py

```diff
@@ -27,6 +27,14 @@ def __init__(self, data=()):
             basetype=Layer,
             lookup={str: lambda e: e.name},
         )
+        self.selection.events.connect(self._on_selection_event)
+
+    def _on_selection_event(self, event):
```

Comment:
I'm hoping that once we drop `layer.selected` and fix the highlight stuff this is no longer needed. Can you add a comment indicating that this is temporary and that it can be dropped with `layer.selected`